aws_lambda_powertools/utilities/data_classes/s3_batch_operation_event.py

```diff
@@ -0,0 +1,220 @@
+        """
+        if self.s3_bucket_arn:
+            return self.s3_bucket_arn.split(":::")[-1]
+        return self["s3Bucket"]
+
+
+class S3BatchOperationEvent(DictWrapper):
+    """Amazon S3BatchOperation Event
+
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/AmazonS3/latest/userguide/batch-ops-invoke-lambda.html
+    """
+
+    @property
+    def invocation_id(self) -> str:
+        """Get the identifier of the invocation request"""
+        return self["invocationId"]
+
+    @property
+    def invocation_schema_version(self) -> str:
+        """ "
+        Get the schema version for the payload that Batch Operations sends when invoking an
+        AWS Lambda function. Either '1.0' or '2.0'.
+        """
+        return self["invocationSchemaVersion"]
+
+    @property
+    def tasks(self) -> Iterator[S3BatchOperationTask]:
+        for task in self["tasks"]:
+            yield S3BatchOperationTask(task)
+
+    @property
+    def task(self) -> S3BatchOperationTask:
+        """Get the first s3 batch operation task"""
+        return next(self.tasks)
+
+    @property
+    def job(self) -> S3BatchOperationJob:
+        """Get the s3 batch operation job"""
+        return S3BatchOperationJob(self["job"])
+
+
+# list of valid result code. Used both in S3BatchOperationResult and S3BatchOperationResponse
+VALID_RESULT_CODE_TYPES: Tuple[str, str, str] = ("Succeeded", "TemporaryFailure", "PermanentFailure")
+
+
+@dataclass(repr=False, order=False)
+class S3BatchOperationResult:
+    task_id: str
+    result_code: Literal["Succeeded", "TemporaryFailure", "PermanentFailure"]
+    result_string: Optional[str] = None
+
+    def __post_init__(self):
+        if self.result_code not in VALID_RESULT_CODE_TYPES:
+            raise ValueError(f"Invalid result_code: {self.result_code}")
+
+    def asdict(self) -> Dict[str, Any]:
+        return {
+            "taskId": self.task_id,
+            "resultCode": self.result_code,
+            "resultString": self.result_string,
+        }
+
+    @classmethod
+    def as_succeeded(cls, task: S3BatchOperationTask, result_string: Optional[str] = None) -> "S3BatchOperationResult":
+        """Create a `Succeeded` result for a given task"""
+        return S3BatchOperationResult(task.task_id, "Succeeded", result_string)
+
+    @classmethod
+    def as_permanent_failure(
+        cls,
+        task: S3BatchOperationTask,
+        result_string: Optional[str] = None,
+    ) -> "S3BatchOperationResult":
+        """Create a `PermanentFailure` result for a given task"""
+        return S3BatchOperationResult(task.task_id, "PermanentFailure", result_string)
+
+    @classmethod
+    def as_temporary_failure(
+        cls,
+        task: S3BatchOperationTask,
+        result_string: Optional[str] = None,
+    ) -> "S3BatchOperationResult":
+        """Create a `TemporaryFailure` result for a given task"""
+        return S3BatchOperationResult(task.task_id, "TemporaryFailure", result_string)
```

Comment:
I'm unsure about this experience. I think we should have the same experience as we have with Kinesis Firehose response. My idea is something like this:

```python
def lambda_handler(event:S3BatchOperationEvent, context: LambdaContext):
    response = S3BatchOperationResponse()
    ...
    result = event.build_s3_batch_response("Succeeded", "This is ok")
    response.add_result(result)
    return response.asdict()
    ...
```

We can get the `task`, `executionid`, and `invocation_id` of the original event and reduce the number of lines of code the customer needs to write.

Make sense for you?